src/general/DayNightCycle.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using Godot;
+using Newtonsoft.Json;
+
+/// <summary>
+///   Controller for variation in sunlight during an in-game day for the current game.
+/// </summary>
+[JsonObject(IsReference = true)]
+[UseThriveSerializer]
+public class DayNightCycle
+{
+    /// <summary>
+    ///   Configuration details for the day/night cycle.
+    /// </summary>
+    [JsonProperty]
+    public DayNightConfiguration LightCycleConfig;
```

Comment:
Do these have to be public *fields*? This is not very good encapsulation design for this class that actually does something. Public properties should only really exist in data classes (or Godot NodePaths as that style became dominant there).